ts_src/psbt.ts

```diff
@@ -810,6 +782,40 @@ export class Psbt {
   }
 }
 
+/**
+ * Helper method for converting a normal Signer into a Taproot Signer.
+ * Note that this helper method requires the Private Key of the Signer to be present.
+ * Steps:
+ *  - if the Y coordinate of the Signer Public Key is odd then negate the Private Key
+ *  - tweak the private key with the provided hash (should be empty for key-path spending)
+ * @param signer - a taproot signer object, the Private Key must be present
+ * @param opts - tweak options
+ * @returns a Signer having the Private and Public keys tweaked
+ */
+export function tweakSigner(signer: Signer, opts: TaprootSignerOpts): Signer {
```

Comment:
This helper function is out of place. It also is returning ECPair which we removed from the library in v6.

It would be much easier to create a new interface called TweakedSigner which has a method called `signTweaked(hash: Buffer, tweak: Buffer, extraEntropy?: Buffer): Buffer` etc.

Then we can implement that for ECPair and bip32 separately.

Reply:
Although, another way to deal with it is to just tell the Signer it is to sign itself tweaked... and just automatically do that with signSchnorr when the internal flag is set... (it knows the tweak since the "recommended" tweak is of its own xonlypubkey)

I don't know which way is more useable... but this helper function is not the right path imo.